doc/classes/String.xml

```diff
@@ -870,35 +952,47 @@
 		<method name="uri_decode" qualifiers="const">
 			<return type="String" />
 			<description>
-				Decodes a string in URL encoded format. This is meant to decode parameters in a URL when receiving an HTTP request.
+				Decodes the string from its URL encoded format. This method is meant to properly decode the parameters in a URL, when receiving an HTTP request.
 				[codeblocks]
 				[gdscript]
-				print("https://example.org/?escaped=" + "Godot%20Engine%3A%27docs%27".uri_decode())
+				var encoded_url = "$DOCS_URL/?highlight=Godot%20Engine%3%docs"
+				var decoded_url = encoded.uri_decode()
+
+				# decoded_url is "$DOCS_URL/?hightlight=Godot Engine:docs"
 				[/gdscript]
 				[csharp]
-				GD.Print("https://example.org/?escaped=" + "Godot%20Engine%3a%27Docs%27".URIDecode());
+				var encoded_url = "$DOCS_URL/?highlight=Godot%20Engine%3%docs"
+				var decoded_url = encoded.uri_decode()
+
+				# decoded_url is "$DOCS_URL/?hightlight=Godot Engine:docs"
 				[/csharp]
 				[/codeblocks]
 			</description>
 		</method>
 		<method name="uri_encode" qualifiers="const">
 			<return type="String" />
 			<description>
-				Encodes a string to URL friendly format. This is meant to encode parameters in a URL when sending an HTTP request.
+				Encodes the string to URL friendly format. This method is meant to properly encode the parameters in a URL, when sending an HTTP request.
```

Comment:
Add a hyphen between URL and friendly. The added comma is unnecessary.

Are there situations where this method fails? Why would it say "This method is meant to properly encode the parameters" and not just "This method encodes the parameters"?

Reply:
The method doesn't fail, but encoding parts of an URL is done to first avoid conflicts with the special characters with further meaning in a string, second, not all characters are valid in an URL because they need to be within the ASCII range. There's no point to encode the whole URL because the "/" slashes will be interpreted properly, regardless.